Test.hs

```diff
@@ -25,61 +25,69 @@ main = do
   where
 
 testShowInstance :: IO ()
-testShowInstance = case a == (show b) of
-  True -> putStrLn "-Pangraph show instance passed."
-  False -> error $ "!Show instance failed test: " ++ show b ++ "!=" ++ a
+testShowInstance = case literal == show graph of
+  True -> putStrLn "-Instance passed"
+  False -> error $ "!Show instance failed test: \"" ++ literal ++ "\"\nSample(above) != Graph(below)\n\"" ++ show graph ++ "\""
   where
-    a = fst showInstanceTestCase
-    b = snd showInstanceTestCase
-    showInstanceTestCase :: (String, P.Pangraph)
-    showInstanceTestCase = ("Pangraph [Node [Attribute (\"id\",\"0\")],Node [Attribute (\"id\",\"1\")]] [Edge [Attribute (\"source\",\"0\"),Attribute (\"target\",\"1\")]]",
-     P.makePangraph [P.makeNode [P.makeAttribute ("id","0")],P.makeNode [P.makeAttribute ("id","1")]] [P.makeEdge [P.makeAttribute ("source","0"),P.makeAttribute ("target","1")]])
+    literal = fst showSample
+    graph = snd showSample
+    showSample :: (String, P.Pangraph)
+    showSample = ("makePangraph [makeVertex \"0\" [(\"id\",\"0\")],makeVertex \"1\" [(\"id\",\"1\")]] [makeEdge [(\"source\",\"0\"),(\"target\",\"1\")] (makeVertex \"0\" [(\"id\",\"0\")],makeVertex \"1\" [(\"id\",\"1\")])]", sampleGraph)
+    sampleGraph :: P.Pangraph
+    sampleGraph =
+      let graph' = P.makePangraph sampleVertices [P.makeEdge [("source","0"), ("target","1")] (sampleVertices !! 0, sampleVertices !! 1)]
+      in  case graph' of
+            Just pangraph -> pangraph
+            Nothing -> error $ "Sample graph failed to build"
+
+    sampleVertices :: [P.Vertex]
+    sampleVertices =
+      [P.makeVertex "0" [("id","0")],
+       P.makeVertex "1" [ ("id","1")]]
     -- Without escapes:
-    -- Pangraph [Node [Attribute ("id","0")],Node [Attribute ("id","1")]] [Edge [Attribute ("source","0"),Attribute ("target","1")]]
+    -- makePangraph [makeVertex "0" [("id","0")],makeVertex "1" [("id","1")]] [makeEdge [("source","0"),("target","1")] (makeVertex "0" [("id","0")],makeVertex "1" [("id","1")])]
 
 testGraphML :: IO ()
 testGraphML = do
-  file <- fmap pack $ readFile path
-  case (parser (head GraphML_P.template) file) of
-    Left x -> error (show x)
-    Right y -> if y == graphs
-      then putStrLn "-Parse Passed"
-      else error $ "!Test failed\n" ++ show y
+  file <- BS.readFile path
+  if graph == parser file
```

Comment:
When I read monadic code I usually prefer to see where things come from immediately, without having to look for the `where` clause. Why don't you define what `graph` is right here, using a `let graph = ...` clause? It would make this function much easier to read.